packages/calcite-components/src/components/block/block.scss

```diff
@@ -3,21 +3,42 @@
  *
  * These properties can be overridden using the component's tag as selector.
  *
- * @prop --calcite-block-padding: Specifies the padding of the block `default` slot.
+ * @prop --calcite-block-background-color: Specifies the background color of the component.
+ * @prop --calcite-block-border-color: Specifies the border color of the component.
+ * @prop --calcite-block-description-text-color: Specifies the text color of the component's description.
+ * @prop --calcite-block-header-background-color: Specifies the background color of the component's header.
+ * @prop --calcite-block-heading-text-color: Specifies the text color of the component's heading.
+ * @prop --calcite-block-padding: **deprecated** use `--calcite-block-space` instead - Specifies the padding of the component's `default` slot.
+ * @prop --calcite-block-space: Specifies the space of the component's `default` slot.
```

Comment:
I think `padding` here is more descriptive - space is a bit ambiguous. Maybe `calcite-block-content-padding` ?

Reply:
I think spacing makes more sense. That way we could internally swap between padding/margin/whatever if need be. Do we want to be specific with padding or margin?

Reply:
We should decide on space vs spacing too :)

Reply:
Yeah, even `content-space` could make sense I guess. But `space` could be anything - space between text, slots, etc. In Modal we do use `--calcite-modal-content-padding`. In the future "content areas" could be ::parts with defaults.

Reply:
We already have spacing vars like `--calcite-spacing-xxs` so it only makes sense to have component spacing vars like `--calcite-block-spacing-` IMO.

Reply:
For Panel, we added `calcite-panel-footer-space`

Reply:
`calcite-block-content-space` works for me. We can update elsewhere we currently use `padding` to follow this. As long as it has a context and isn't just `component-space`.

Reply:
I knew this seemed familiar. We had a similar discussion in https://github.com/Esri/calcite-design-system/pull/8629#discussion_r1463970044 and landed on `--calcite-tab-content-block-padding`. 

> `calcite-block-content-space` works for me. We can update elsewhere we currently use padding to follow this. As long as it has a context and isn't just component-space.

SGTM, but I'd like for @alisonailea to chime on this too. If we needed separate props later on for block/inline values, would we introduce `--calcite-component-space-<block|inline>`? If so, we could add an example of this in our component token guidelines.

> We should decide on space vs spacing too :)

We decided on `space` to align with `size` (vs. `sizing`).

Reply:
Given that our design system is not necessarily web specific, we need to use platform agnostic naming conventions.   `space` is the preferred terminology here.

Reply:
For posterity, if we need custom space props that only apply horizontally or vertically, we can use something like `--calcite-component-<part>-space-<x|y>` (see https://github.com/Esri/calcite-design-system/pull/8852#discussion_r1508358308).